tasty-query/shared/src/main/scala/tastyquery/reader/tasties/TreeUnpickler.scala

```diff
@@ -1095,6 +1084,48 @@ private[tasties] class TreeUnpickler(
       throw TastyFormatException(s"Unexpected type tag ${astTagToString(tag)} $posErrorMsg")
   }
 
+  private def readLambdaType[N <: Name, PInfo <: Type | TypeBounds, LT <: LambdaType](
+    companionOp: FlagSet => LambdaTypeCompanion[N, PInfo, LT],
+    nameMap: TermName => N,
+    readInfo: TreeUnpickler => PInfo
+  )(using LocalContext): LT =
+    val lambdaAddr = reader.currentAddr
+    reader.readByte()
+    val end = reader.readEnd()
+
+    val resultUnpickler = fork // remember where the result type is
+    skipTree() // skip the result type
+    val paramInfosUnpickler = fork // remember where the params are
+
+    // Read names -- skip infos, and stop if we find a modifier
+    val paramNames = reader.collectWhile(reader.currentAddr != end && !isModifierTag(reader.nextByte)) {
+      skipTree()
+      nameMap(readName)
+    }
+
+    // Read mods
+    var mods = EmptyFlagSet
+    while reader.currentAddr != end do // avoid boxing the mods
+      reader.readByte() match
+        case IMPLICIT => mods |= Implicit
+        case ERASED   => mods |= Erased
+        case GIVEN    => mods |= Given
+
+    // Read infos -- skip names
+    def readParamInfos()(using LocalContext): List[PInfo] =
+      paramInfosUnpickler.reader.until(end) {
```

Comment:
this looks like it would try to readInfo when the next byte is the mods

Reply:
this example causes `unexpected type tag GIVEN` error:
```scala
val givenRefinement = ??? : { def foo(using Int): Int }
```

Reply:
Oh nice catch! It's fixed now.